shell/shared/db.js

```diff
@@ -86,6 +86,46 @@ Grains = new Mongo.Collection("grains");
 //   publicId:  An id used to publicly identify this grain. Used e.g. to route incoming e-mail and
 //       web publishing. This field is initialized when first requested by the app.
 
+RoleAssignments = new Mongo.Collection("roleAssignments");
+// A role assignment allows a user to access a grain with a specified set of permissions; it is an
+// edge in the permissions sharing graph. A user's permissions for a given grain are computed as
+// the union of the permissions received through role assignments.
+//
+// Any grain for which a user has received a role assignment will show up in that user's grain list.
+// Role assignments are also used to restrict access during restoration of persisted capabilities.
+//
+// Each contains:
+//   _id: random
+//   grainId: The `_id` of the grain whose permissions are being shared.
+//   sharer: The `_id` of the user who is sharing these permissions.
+//   recipient: The `_id` of the user who receives these permissions.
+//   permissions: An array of bools representing the received permissions.
```

Comment:
It might make sense to define this as being "bonus" permissions in addition to what the role assignment grants. This way an app can change the mapping of roles to permissions over time. In fact the only reason to have the `permissions` field in this case is if we want to offer an "advanced" UI in which users can select individual permissions, but I'd argue that there's no immediate need to implement, so we should just leave out `permissions` entirely and go based on `roleAssignment`.